syntax/str_stdlib_test.go

```diff
@@ -0,0 +1,175 @@
+package syntax
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSub(t *testing.T) {
+	t.Parallel()
+
+	AssertCodesEvalToSameValue(
+		t,
+		`"this is a test"`,
+		`//.str.sub("this is not a test", "is not", "is")`,
+	)
+	AssertCodesEvalToSameValue(
+		t,
+		`"this is still a test"`,
+		`//.str.sub("this is still a test", "doesn't matter", "hello there")`,
+	)
+	assertExprPanics(t, `//.str.sub("hello there", "test", 1)`)
+}
+
+func TestSplit(t *testing.T) {
+	t.Parallel()
+
+	AssertCodesEvalToSameValue(
+		t,
+		`["this", "is", "a", "test"]`,
+		`//.str.split("this is a test", " ")`,
+	)
+	AssertCodesEvalToSameValue(
+		t,
+		`["this is a test"]`,
+		`//.str.split("this is a test", ",")`,
+	)
+	AssertCodesEvalToSameValue(
+		t,
+		`["th", " ", " a test"]`,
+		`//.str.split("this is a test", "is")`,
+	)
```

Comment:
One-liners are much more readable:
```go
AssertCodesEvalToSameValue(t, `["this", "is", "a", "test"]`, `//.str.split("this is a test", " ")`)
AssertCodesEvalToSameValue(t, `["this is a test"]         `, `//.str.split("this is a test", ",")`)
AssertCodesEvalToSameValue(t, `["th", " ", " a test"]     `, `//.str.split("this is a test", "is")`)
```

I generally have a hard-limit of 120 columns. Within that, it's a judgment call.